pkg/middleware/middleware.go

```diff
@@ -608,7 +608,7 @@ func (s *SouinBaseHandler) ServeHTTP(rw http.ResponseWriter, rq *http.Request, n
 				}
 				customWriter.WriteHeader(response.StatusCode)
 				s.Configuration.GetLogger().Sugar().Debugf("Serve from cache %+v", req)
-				_, _ = io.Copy(customWriter.Buf, response.Body)
+				_, _ = io.Copy(customWriter, response.Body)
```

Comment:
Why not copying the buffer?

Reply:
Because otherwise we don't benefit from the locking performed on `customWriter#Write`